Source/WebCore/platform/network/cocoa/WebCoreNSURLSession.mm

```diff
@@ -897,11 +901,11 @@ - (void)resume
 
 - (void)dealloc
 {
-    if (_resource) {
-        _targetQueue->dispatch([client = _resource->client(), resource = WTFMove(_resource)] () mutable {
-            resource->shutdown();
-            if (client)
+    if (RefPtr<PlatformMediaResource> resource = std::exchange(_resource, { })) {
```

Comment:
Oh I see, dealloc is not guaranteed in target queue so _resource is also accessed elsewhere potentially.

Reply:
but then we have no longer a reference to `_resource` member, so it can't be accessed concurrently from elsewhere.

Not unless ObjC's dealloc isn't thread-safe.